src/components/app/AppViewRouter.js

```diff
@@ -81,24 +81,43 @@ class AppViewRouterImpl extends PureComponent<AppViewRouterProps> {
       case 'DATA_RELOAD':
         return <ProfileLoaderAnimation />;
       case 'FATAL_ERROR': {
-        const message =
-          ERROR_MESSAGES_L10N_ID[dataSource] ||
-          'AppViewRouter--error-message-public';
+        let message =
+          ERROR_MESSAGES_L10N_ID[dataSource] || 'AppViewRouter--error-public';
         let additionalMessage = null;
         if (view.error) {
-          console.error(view.error);
-          additionalMessage =
-            `${view.error.toString()}\n` +
-            'The full stack has been written to the Web Console.';
+          if (view.error.name === 'SafariLocalhostHTTPLoadError') {
+            message = 'AppViewRouter--error-from-localhost-url-safari';
+          } else {
+            console.error(view.error);
+            additionalMessage = (
+              <>
+                <p>{view.error.toString()}</p>
+                <p>The full stack has been written to the Web Console.</p>
```

Comment:
Next step will be localizing this :) feel free to do it if you feel like it, but otherwise we can do it later

Reply:
I'm not planning on doing that in the near future.